src/settings/rageshake.ts

```diff
@@ -239,7 +237,7 @@ class IndexedDBLogStore {
       }
       // queue up a flush to occur immediately after the pending one completes.
       this.flushAgainPromise = this.flushPromise
-        .then(() => {
+        .then(async () => {
           return this.flush();
         })
```

Comment:
Nitpick:
```suggestion
        .then(this.flush)
```
which is safe as long as `flush` is an arrow function (and thus gets the correct binding for its `this`).